zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+        flask.abort(401)
+    return jsonify(get_user_info(id))
+
+# Gets user words info
+@with_session
+def get_user_info(id):
+    try:
+        user = User.query.filter_by(id=id).one()
+        dictionary = {
+            'id': str(id),
+            'name': user.name,
+            'email': user.email,
+            'reading_time': random.randint(1, 100),
+            'exercises_done': random.randint(1, 100),
+            'last_article': 'place holder article'
+        }
+        return dictionary
+    except ValueError:
+        flask.abort(400)
+
+
+
+
+# Removes class
+@api.route("/remove_class/<class_id>", methods=["POST"])
+@with_session
+def remove_class(class_id):
+    from zeeguu.model import TeacherCohortMap
+    if(not class_function_checker(class_id)):
+        flask.abort(401)
+    try:
+        selected_cohort = Cohort.query.filter_by(id=class_id).one()
+
+        if not selected_cohort.cur_students == 0:
+            flask.abort(400)
+
+        links = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+        for link in links:
+            zeeguu.db.session.delete(link)
+        zeeguu.db.session.delete(selected_cohort)
+        zeeguu.db.session.commit()
+        return 'removed'
+    except ValueError:
+        flask.abort(400)
+
+# Takes Teacher id as input and outputs list of all cohort_ids that teacher owns
+@api.route("/get_classes", methods=["GET"])
+@with_session
+def get_classes_by_teacher_id():
+    from zeeguu.model import TeacherCohortMap
+    mappings = TeacherCohortMap.query.filter_by(user_id=flask.g.user.id).all()
+    cohorts = []
+    for m in mappings:
+        info = get_class_info(m.cohort_id)
+        cohorts.append(info)
+    return json.dumps(cohorts)
+
+
+# Takes cohort_id and reuturns dictionary with relevant class variables
+@api.route("/get_class_info/<id>", methods=["GET"])
+@with_session
+def wrapper_to_json_class(id):
+    if(not class_function_checker(id)):
+        flask.abort(401)
+    return jsonify(get_class_info(id))
+
+
+def get_class_info(id):
+    c = Cohort.find(id)
+    class_name = c.class_name
+    inv_code = c.inv_code
+    max_students = c.max_students
+    cur_students = c.cur_students
+    class_language_id = c.class_language_id
+    class_language = Language.query.filter_by(id=class_language_id).one()
+    d = {'id': str(id), 'class_name': class_name, 'inv_code': inv_code, 'max_students': max_students,
+         'cur_students': cur_students, 'class_language_name': class_language.name, 'class_id': id}
+    return d
+
+
+# Takes two inputs (user_id, cohort_id) and links them other in teacher_cohort_map table.
+# url input in format <user_id>/<cohort_id>
+#@api.route("/link_teacher_class/<user_id>/<cohort_id>", methods=["POST"])
+def link_teacher_class(user_id, cohort_id):
+    from zeeguu.model import TeacherCohortMap
+    user = User.find_by_id(user_id)
+    cohort = Cohort.find(cohort_id)
+    zeeguu.db.session.add(TeacherCohortMap(user, cohort))
```

Comment:
you've already imported db in the scope, so you can say db.sesison(...)

Reply:
Replaced all references of zeeguu.db to just db. Defining db as zeeguu.db at the top of the file.